src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -46,6 +47,53 @@
     "flags": ([str], "List of active flags")
 }
 
+device_status_template = {
+    'options': {
+        'cost-capacity': {
+            'name': 'cost-capacity',
+            'available': False,
+            'capacity': 0,
+            'used': 0,
+            'hdd_used': 0,
+            'ssd_used': 0,
+            'nvme_used': 0,
+        },
+        'throughput': {
+            'name': 'throughput-optimized',
+            'available': False,
+            'capacity': 0,
+            'used': 0,
+            'hdd_used': 0,
+            'ssd_used': 0,
+            'nvme_used': 0,
+        },
+        'iops': {
+            'name': 'iops-optimized',
+            'available': False,
+            'capacity': 0,
+            'used': 0,
+            'hdd_used': 0,
+            'ssd_used': 0,
+            'nvme_used': 0,
+        },
+    },
+    'recommended_option': None
+}
```

Comment:
While dictionary and JSONs are great  to prototype, you should always try to convert those into data structures. The three scenarios here share a common structure:
```py
class OSDDeployment(NamedTuple):
  name: str
  available: bool = False
  capacity: int = 0
  used: int = 0
  hdd_used: int = 0
  ssd_used: int = 0
  nvme_used: int = 0
```
 BTW, JSONs and dicts are extremely prone to error. If you write `nvme_used` in one place, and `nmve_used` in another, you won't easily spot the error (and you won't get any warning, since both are valid, they're just text). That's why we should move quickly to structured types after first prototyping pass.